subsys/nfc/lib/platform.c

```diff
@@ -64,46 +65,55 @@ nrfx_err_t nfc_platform_setup(void)
 }
 
 
-static nrfx_err_t nfc_platform_tagheader_get(uint8_t index,
-					     uint32_t *tag_header)
+static nrfx_err_t nfc_platform_tagheaders_get(uint32_t tag_header[3])
 {
-	const uint32_t tag_header_addresses[] = {
-#ifdef CONFIG_TRUSTED_EXECUTION_NONSECURE
-		(uint32_t) &NRF_FICR_S->NFC.TAGHEADER0,
-		(uint32_t) &NRF_FICR_S->NFC.TAGHEADER1,
-		(uint32_t) &NRF_FICR_S->NFC.TAGHEADER2
+	__IOM FICR_NFC_Type *ficr_nfc =
+#if defined(NRF_TRUSTZONE_NONSECURE)
+				&NRF_FICR_S->NFC;
 #else
-		(uint32_t) &NRF_FICR->NFC.TAGHEADER0,
-		(uint32_t) &NRF_FICR->NFC.TAGHEADER1,
-		(uint32_t) &NRF_FICR->NFC.TAGHEADER2
+				&NRF_FICR->NFC;
 #endif
-	};
 
-	if (index >= ARRAY_SIZE(tag_header_addresses)) {
-		LOG_ERR("Tag Header index out of bounds");
-		return NRFX_ERROR_INVALID_ADDR;
-	}
 
 #ifdef CONFIG_TRUSTED_EXECUTION_NONSECURE
+#if defined(CONFIG_BUILD_WITH_TFM)
+	uint32_t err = 0;
+	enum tfm_platform_err_t plt_err;
+	FICR_NFC_Type ficr_nfc_ns;
+
+	plt_err = tfm_platform_mem_read(&ficr_nfc_ns, (uint32_t)ficr_nfc,
```

Comment:
s/ficr_nfc/&NRF_FICR_S->NFC/

ficr_nfc is very closely named to ficr_nfc_ns, which can be confusing to a tired reader.

I don't see the need for the ficr_nfc abstraction. It's more obvious what &NRF_FICR_S->NFC is than what ficr_nfc is.

Reply:
I did this because then the common code that uses the pointer can be outside the ifdefs.
Otherwise I would have duplicate that code inside the ifdefs, which I want to avoid.

Reply:
discussed offline